packages/server/routes/api/guilds.js

```diff
@@ -5,14 +5,40 @@ const GuildController = require("../../controllers/guilds");
 const AuthController = require("../../controllers/auth");
 
 router.get("/", AuthController.authenticate, async (request, response) => {
-  try {
-    const guilds = await GuildController.getAllGuilds();
-    response.status(200).json({
-      status: "success",
+  const search = request.query.search;
+
+  if (!search) {
+    response.status(400).json({
+      status: "fail",
       data: {
-        guilds,
+        search: "Missing search parameter.",
       },
     });
+    return;
+  }
+
+  let guilds;
+  try {
+    if (search.startsWith("@")) {
+      guilds = await GuildController.searchGuildByHandler(search);
+    } else {
+      guilds = await GuildController.searchGuildByName(search);
+    }
+    if (guilds.length === 0) {
+      response.status(404).json({
+        status: "fail",
+        data: {
+          search: "A guild was not found in the query.",
+        },
+      });
```

Comment:
A more descriptive failure message would be good like `"A guild was not found with the query ${search}"`.